src/compile/decode.js

```diff
@@ -132,14 +198,17 @@ function compileDecode (m, resolve, enc) {
 
           if (val) { // is enum
             if (field.repeated) {
-              obj[name] = []
+              def = []
             } else {
               def = (def && val[def]) ? val[def].value : val[Object.keys(val)[0]].value
-              obj[name] = parseInt(def || 0, 10)
+              def = parseInt(def || 0, 10)
```

Comment:
for enums, defaultVal will be undefined even if they are set

Reply:
I think that's ok because they are always set.

In the [`defaults decode` test](https://github.com/ipfs/protons/blob/master/test/defaults.js#L32) - `o1` is decoded from an empty buffer and `foo2` [has no default value](https://github.com/ipfs/protons/blob/master/test/test.proto#L61) so is initialised to the value of the [first entry of the `FOO` enum](https://github.com/ipfs/protons/blob/master/test/test.proto#L54).